be/src/exec/olap_scanner.cpp

```diff
@@ -273,6 +279,13 @@ Status OlapScanner::get_batch(RuntimeState* state, RowBatch* batch, bool* eof) {
                 break;
             }
             // Read one row from reader
+            allocated_bytes = mem_pool->total_reserved_bytes() - allocated_bytes;
+            if (allocated_bytes > 1024 * 1024) {
+                if (_mem_tracker.get()->any_limit_exceeded()) {
```

Comment:
After the Hook tcmalloc new/delete is merged,
We can automatically limit the memory in similar places~, instead of manually